akd/src/directory.rs

```diff
@@ -92,11 +95,21 @@ where
         let current_epoch = current_azks.get_latest_epoch();
         let next_epoch = current_epoch + 1;
 
-        let mut keys: Vec<AkdLabel> = updates
-            .iter()
-            .map(|(akd_label, _val)| akd_label.clone())
-            .collect();
+        let updates_iter = updates.iter().map(|(akd_label, _val)| akd_label.clone());
```

Comment:
I think it might be more idiomatic to do

```rust
let distinct_set = updates
    .iter()
    .map(|(label, _)| label.clone())
    .collect::<HashSet<_>>()

if distinct_set.len() != updates.len() {
    return Err(...);
}
```